google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -1560,6 +1562,46 @@ def delete generation: nil,
           true
         end
 
+        # Mode of object level retention configuration.
+        # Valid values are 'Locked' or 'Unlocked'
+        #
+        # @return [String]
+        def retention_mode
+          @gapi.retention&.mode
+        end
+
+        # The earliest time in RFC 3339 UTC "Zulu" format that the object can
+        # be deleted or replaced.
+        #
+        # @return [DateTime]
+        def retention_retain_until_time
+          @gapi.retention&.retain_until_time
+        end
+
+        # A collection of object level retention parameters.
+        # The full list of available options are outlined at the [JSON API docs]
+        # (https://cloud.google.com/storage/docs/json_api/v1/objects/insert#request-body).
+        #
+        # @return [Google::Apis::StorageV1::Object::Retention]
+        def retention
+          @gapi.retention
+        end
+
+        ##
+        # Update method to update retention parameter of an object / file
+        # It accepts params as a Hash of attributes in the following format:
+        #
+        #     { mode: 'Locked|Unlocked', retain_until_time: '2023-12-19T03:22:23+00:00' }
+        #
+        # @param [Hash(String => String)] new_retention_attributes
```

Comment:
[Optional Nit] This is better read as `@param [Hash(String => String)] new_retention_attributes`.  Has an added benefit of separating parameter names from its description, if we update it in the future.

Reply:
I don't see a difference. What am I missing?

Reply:
YARD treats both of them as the same (hence, the optional nit).

However, the below format makes it easy to separate the parameter from it's description:

```
@param arg_name [datatype] long description text
```

In contrast, the below makes it harder to spot parameter name from description:
```
@param [datatype] arg_name long description text
```

Reply:
Hmm. Makes sense. But this file has other places with the same format.  For e.g.: https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-storage/lib/google/cloud/storage/file.rb#L359-L366